### PR TITLE
Feature/slackdriver

### PR DESCRIPTION
**Pull request by herpaderpaldent**
Implement slackdriver

### CODE REVIEW

File: src/Drivers/SlackNotificationDriver.php [hunk 1]
```diff
@@ -132,7 +134,23 @@ public static function getPrivateChannel() : ?string
      */
     public static function getPublicDriverId(string $notification) : ?string
     {
-        return null;
+        try {
```
Comment (warlof) [inline src/Drivers/SlackNotificationDriver.php]: What will change for another driver?
Can't it be moved to the abstract parent class?

File: src/Http/Controllers/Slack/SlackUserOAuthController.php [hunk 2]
```diff
@@ -86,9 +94,21 @@ public function callback()
                 ->with('error', 'An error occurred while exchanging credentials with slack. ' . $e->getMessage());
         }
 
-        return redirect()->route('seatnotifications.index')
-            ->with('success', 'Now we know who to notify and you can subscribe to any of the available notifications.');
+        $driver = request()->session()->pull('herpaderp.seatnotifications.subscribe.driver');
+        $notification = request()->session()->pull('herpaderp.seatnotifications.subscribe.notification');
+        $characters_filter = request()->session()->pull('herpaderp.seatnotifications.subscribe.characters_filter');
```
Comment (warlof) [inline src/Http/Controllers/Slack/SlackUserOAuthController.php]: The driver should only be annoyed with stuff concerning the way binding and notification are sent.

We should find a way to extract filters from this logic

Comment (herpaderpaldent) [reply src/Http/Controllers/Slack/SlackUserOAuthController.php]: you are right, i copy that from discord auth part. If i read this correctly you had introduced it probably its obsolete:
https://github.com/herpaderpaldent/seat-notifications/blame/23269a36fd8ac8f2f4be9e7fcae6b989bcfd8cbb/src/Http/Controllers/Discord/DiscordOAuthController.php#L101-L102

File: src/Drivers/SlackNotificationDriver.php (hunk 1 above)
Comment (herpaderpaldent) [reply src/Drivers/SlackNotificationDriver.php]: i agree, i'll move it.

File: src/Http/Controllers/Slack/SlackUserOAuthController.php (hunk 2 above)
Comment (warlof) [reply src/Http/Controllers/Slack/SlackUserOAuthController.php]: I didn't say obsolete but that it was a duplicated area ;)

File: src/Http/Controllers/Discord/DiscordOAuthController.php [hunk 3]
```diff
@@ -98,14 +101,10 @@ public function callback()
...
-        $characters_filter = request()->session()->pull('herpaderp.seatnotifications.subscribe.characters_filter');
-        $corporations_filter = request()->session()->pull('herpaderp.seatnotifications.subscribe.corporations_filter');
 
         $data = [
             'driver' => $driver,
             'notification' => $notification,
-            'characters_filter' => $characters_filter,
-            'corporations_filter' => $corporations_filter,
             'driver_id' => DiscordUser::find(auth()->user()->group->id)->channel_id,
             'group_id' => auth()->user()->group->id,
```
Comment (warlof) [inline src/Http/Controllers/Discord/DiscordOAuthController.php]: Can't it be move in a generic place either ?

**Pull request merged**